Validation/SiTrackerPhase2V/plugins/Phase2OTValidateTTStub.cc

```diff
@@ -81,48 +128,279 @@ Phase2OTValidateTTStub::~Phase2OTValidateTTStub() {
 }
 
 void Phase2OTValidateTTStub::dqmBeginRun(const edm::Run &iRun, const edm::EventSetup &iSetup) {
-  tkGeom_ = &(iSetup.getData(geomToken_));
-  tTopo_ = &(iSetup.getData(topoToken_));
+  theTrackerGeom_ = &(iSetup.getData(geomToken_));
+  tTopo_ = &(iSetup.getData(tTopoToken_));
+
+  // Clear existing histograms
+  phi_res_barrel_layers.clear();
+  bend_res_barrel_layers.clear();
+  phi_res_fw_endcap_discs.clear();
+  bend_res_fw_endcap_discs.clear();
+  phi_res_bw_endcap_discs.clear();
+  bend_res_bw_endcap_discs.clear();
+
+  // Resize vectors and set elements to nullptr
+  phi_res_barrel_layers.resize(6, nullptr);
```

Comment:
Having magic numbers in the code like 5 or 6 is not allowed. At the very least these need to be named constants.